app/rest/proxy/avatar_store.go

```diff
@@ -85,3 +101,41 @@ func (fs *FSAvatarStore) location(id string) string {
 	partition := checksum64 % 100
 	return path.Join(fs.storePath, fmt.Sprintf("%02d", partition))
 }
+
+// Resizes an image of supported format (PNG, JPG, GIF) to the size of "limit" px of the biggest side
+// (width or height) preserving aspect ratio.
+// Returns original reader if resizing is not needed or failed.
+func resize(reader io.Reader, limit int) (io.Reader, error) {
+	if limit <= 0 {
+		return reader, errors.New("limit should be greater than 0")
+	}
+
+	var teeBuf bytes.Buffer
+	tee := io.TeeReader(reader, &teeBuf)
+	src, _, err := image.Decode(tee)
+	if err != nil {
+		return &teeBuf, errors.Wrap(err, "can't decode avatar image")
+	}
+
+	bounds := src.Bounds()
+	w, h := bounds.Dx(), bounds.Dy()
+	if w <= limit && h <= limit || w <= 0 || h <= 0 {
+		return &teeBuf, errors.New("resizing image is smaller that the limit or has 0 size")
+	}
+	var newW, newH int
```

Comment:
I don't like else construct, and this part will be simpler without. To me, this little unnecessary calculation justified by better readability. 

```
newW, newH := w*limit/h, limit
if w > h {
	newW, newH = limit, h*limit/w
}
```